feat(bhAccountSelect): filter out locked accounts

jniles opened the pull request:
This commit implements the locked account filter on the bhAccountSelect. It also cleans up the report controller and removes the `classe` variable since it isn't used in any modules that I can find.

Closes #2808.

![2018-06-07_14-21-46](https://user-images.githubusercontent.com/896472/41103286-ac1da27a-6a60-11e8-9509-cc007bbbe22d.gif)

jniles committed:
feat(bhAccountSelect): filter out locked accounts

This commit implements the locked account filter on the bhAccountSelect.
It also cleans up the report controller and removes the classe variable
since it isn't used in any modules that I can find.

Closes #2808.

jniles committed:
feat(accounts): implement hidden api

This commit implements and tests the "hidden" column API.  The hidden
column is an extra column to toggle account visibility throughout the
system, but doesn't have a logical (read: accounting) impact on
functioning of the system.

The preliminary work has also been laid out for the dropdown menu in the
account configuration module.

Partially addresses: #2808.

jniles committed:
feat(accounts): implement actions dropdown

This accounts page now has an actions dropdown with all the options tied
to manipulating accounts.  This includes:
 1. Editing
 2. Deleting
 3. Hiding
 4. Locking

jniles committed:
test(accounts): implement dropdown menu tests

This commit migrates the account tests to use the dropdown menu on
accounts.

mbayopanda commented on client/src/i18n/en/account.json:
```diff
@@ -47,6 +47,10 @@
+    "CONFIRM_HIDE" : "Hiding an account removes the account from all account selection elements.  It will not affect any pending transactions using the account.  Make it hidden?",
+    "CONFIRM_UNHIDE" : "This account is currently hidden. Are you sure you want to reveal it for use?",
+    "CONFIRM_LOCK" : "Locking an account restrict the ability to post transactions using the account.  Are you sure you want to lock it?",
+    "CONFIRM_UNLOCK" : "The account is locked.  Unlocking it will allow transactions to be posted containing this account.  Are you sure you want to unlock it?",
```
There are missing french translations.
```js
"CONFIRM_HIDE" : "Masquer un compte supprime le compte de tous les éléments de sélection de compte. Cela n'affectera pas les transactions en attente utilisant le compte. Le rendre caché ?",
"CONFIRM_UNHIDE" : "Ce compte est actuellement caché. Êtes-vous sûr de vouloir le révéler pour l'utiliser ?",
"CONFIRM_LOCK" : "Le verrouillage d'un compte limite la possibilité de publier des transactions en utilisant le compte. Êtes-vous sûr de vouloir le verrouiller ?",
"CONFIRM_UNLOCK" : "Le compte est bloqué. Le débloquer permettra d'afficher des transactions contenant ce compte. Êtes-vous sûr de vouloir le débloquer ?",
```

jniles replied:
Ah, beautiful.  This is exactly what I needed!

mbayopanda commented on client/src/modules/accounts/templates/grid.actionsCell.tmpl.html:
```diff
@@ -1,8 +1,64 @@
-    <i class="fa fa-edit"></i>
-    <span translate>TABLE.COLUMNS.EDIT</span> {{row.entity.number}}
+<div class="ui-grid-cell-contents" uib-dropdown dropdown-append-to-body data-action="{{ row.entity.id }}">
+  <a href uib-dropdown-toggle>
```
👍 nice

mbayopanda commented on server/controllers/finance/accounts/index.js:
```diff
@@ -32,11 +32,10 @@ const types = require('./types');
 const categories = require('./categories');
 const Periods = require('../../../lib/period');
 const AccountExtras = require('./extra.js');
```
`./extra.js` could be also `./extra`

jniles replied:
Will do.

mbayopanda reviewed (approved):
LGTM

jniles committed:
fix(accounts): add i18n, client hidden filter

This commit adds French translations (thanks @mbayopanda) and makes sure
the "hidden" filter is applied on the client-side instead of locked.
